gateway/xgress_edge/listener.go

```diff
@@ -150,13 +150,32 @@ func (proxy *ingressProxy) processConnect(req *channel2.Message, ch channel2.Cha
 	log.Debug("dialing fabric")
 	peerData := make(map[uint32][]byte)
 	peerData[edge.PublicKeyHeader] = req.Headers[edge.PublicKeyHeader]
+
+	if ns.Service.EncryptionRequired && req.Headers[edge.PublicKeyHeader] == nil {
```

Comment:
`conn.close(true, msg)` sends a state close reply, so you don't need both. I think we should be doing `conn.close(true, msg)` in the block above as well (if session create fails)

Reply:
There is a difference.

`conn.close` doesn't reply to the incoming `req` message. The `sendStateClosedReply` does by calling `replyFor()` on the state closed message.

Hence my confusion as in the original code both are used.

Reply:
Talked w/ Paul. Should `sendStateClosedReply` then `conn.close` w/o notify.